src/OrchardCore/OrchardCore.ResourceManagement/ResourceManager.cs

```diff
@@ -308,46 +314,53 @@ private bool ResolveInlineDefinitions(string resourceType)
 
         private IEnumerable<RequireSettings> ResolveRequiredResources(string resourceType)
         {
-            return _required.Where(r => r.Key.Type == resourceType).Select(r => r.Value);
-        }
-
-        public IEnumerable<LinkEntry> GetRegisteredLinks()
-        {
-            if (_links == null)
+            foreach (var (key, value) in _required)
             {
-                return Enumerable.Empty<LinkEntry>();
+                if (key.Type == resourceType)
+                {
+                    yield return value;
+                }
             }
-
-            return _links.AsReadOnly();
         }
 
-        public IEnumerable<MetaEntry> GetRegisteredMetas()
+        public IEnumerable<LinkEntry> GetRegisteredLinks() => DoGetRegisteredLinks();
```

Comment:
here's public API + DoAPI, Do prefix uses concrete types and is faster when called inside the class, public API exposes IEnumerable which is slowest to traverse (.NET Core has made huge progress here checking the type behind IEnumerable, but still).